### ISSUE TITLE
[master] Opening user management triggers skeletion copy on every user!

### ISSUE THREAD

**Issue by DeepDiver1975** [owncloud/core#22119]
````
[Thu Feb  4 10:35:11 2016] copying skeleton for bazbarfoo from /home/deepdiver/Development/ownCloud/master/core/skeleton to /bazbarfoo/files/
[Thu Feb  4 10:35:12 2016] rewinddir(): 46 is not a valid Directory resource at /home/deepdiver/Development/ownCloud/master/3rdparty/icewind/streams/src/Wrapper.php#31
[Thu Feb  4 10:35:12 2016] copying skeleton for foobarbaz from /home/deepdiver/Development/ownCloud/master/core/skeleton to /foobarbaz/files/
[Thu Feb  4 10:35:13 2016] rewinddir(): 59 is not a valid Directory resource at /home/deepdiver/Development/ownCloud/master/3rdparty/icewind/streams/src/Wrapper.php#31
[Thu Feb  4 10:35:13 2016] copying skeleton for notme from /home/deepdiver/Development/ownCloud/master/core/skeleton to /notme/files/
[Thu Feb  4 10:35:14 2016] rewinddir(): 72 is not a valid Directory resource at /home/deepdiver/Development/ownCloud/master/3rdparty/icewind/streams/src/Wrapper.php#31
[Thu Feb  4 10:35:14 2016] copying skeleton for user0 from /home/deepdiver/Development/ownCloud/master/core/skeleton to /user0/files/
[Thu Feb  4 10:35:15 2016] rewinddir(): 85 is not a valid Directory resource at /home/deepdiver/Development/ownCloud/master/3rdparty/icewind/streams/src/Wrapper.php#31
[Thu Feb  4 10:35:16 2016] copying skeleton for user01 from /home/deepdiver/Development/ownCloud/master/core/skeleton to /user01/files/
[Thu Feb  4 10:35:17 2016] rewinddir(): 98 is not a valid Directory resource at /home/deepdiver/Development/ownCloud/master/3rdparty/icewind/streams/src/Wrapper.php#31
[Thu Feb  4 10:35:17 2016] copying skeleton for user02 from /home/deepdiver/Development/ownCloud/master/core/skeleton to /user02/files/
[Thu Feb  4 10:35:18 2016] rewinddir(): 111 is not a valid Directory resource at /home/deepdiver/Development/ownCloud/master/3rdparty/icewind/streams/src/Wrapper.php#31
[Thu Feb  4 10:35:18 2016] copying skeleton for user_2T55Anbh3MmrX from /home/deepdiver/Development/ownCloud/master/core/skeleton to /user_2T55Anbh3MmrX/files/
[Thu Feb  4 10:35:19 2016] rewinddir(): 124 is not a valid Directory resource at /home/deepdiver/Development/ownCloud/master/3rdparty/icewind/streams/src/Wrapper.php#31
[Thu Feb  4 10:35:19 2016] copying skeleton for user_kh0xRolXH7Xmt from /home/deepdiver/Development/ownCloud/master/core/skeleton to /user_kh0xRolXH7Xmt/files/
[Thu Feb  4 10:35:20 2016] rewinddir(): 137 is not a valid Directory resource at /home/deepdiver/Development/ownCloud/master/3rdparty/icewind/streams/src/Wrapper.php#31

**Comment by PVince81** [comment]
Ouch. Possible regression ? Let me check 8.2

**Comment by DeepDiver1975** [comment]
getting the avatar for each user initialized this user's data dir which triggers the skeletion copy

**Comment by DeepDiver1975** [comment]
![bildschirmfoto von 2016-02-04 11-52-30](https://cloud.githubusercontent.com/assets/1005065/12813023/cfa35d10-cb35-11e5-8bbf-cc234af5ab43.png)

**Comment by PVince81** [comment]
Hmm, then it's possibly older.

CC @rullzer

**Comment by PVince81** [comment]
Maybe the avatar code should not do anything if the user's home was initialized, in which case it can mean that the user don't have an avatar yet. But then, there's LDAP avatars...

**Comment by PVince81** [comment]
Setting to high, could impair user management performance.

@DeepDiver1975 does it only do it once for users created on the CLI or does it do this **every** time the settings page is opened ? (if every time => high, if only once => medium)

**Comment by rullzer** [comment]
It basically does this for each 'unitialized' user.

Avatar code requestst the UserFolder
When the user folder is requestsed we see if it is there
For unitisialied users it is not
We create it and copy the skeleton

So basically this is happening because the avatar code now is using the node api...

**Comment by PVince81** [comment]
And the avatar needs access to the user's filesystem because sometimes the avatar is stored there ?

Is it possible to first detect whether the user has an avatar in the first place **before** calling `getUserFolder()` ? (and goodbye DI for that item)

**Comment by PVince81** [comment]
another alternative is to have an option "don't create skeleton" for `getUserFolder()`, but that sounds a bit far-fetched. @icewind1991

**Comment by DeepDiver1975** [comment]
The basic question is if the copy skeleton operation is executed in the correct location?

Generally speaking my expectation of a get method would be to not change anything.

Should we copy the skeletion on login? on create of the user? (issue with shibboleth)

**Comment by rullzer** [comment]
@PVince81 basically the avatar is just a file stored in <USER>/avatar.jpg

**Comment by PVince81** [comment]
Hmm, so there is no oc_preferences entry that tells so ? Because it needs to know the file extension too ?

**Comment by rullzer** [comment]
Nope we just check both for existence

**Comment by rullzer** [comment]
Not saying it is ideal... just that this is how we do it ;)

**Comment by DeepDiver1975** [comment]
maybe we need to retink the avatar storage location ....

**Comment by rullzer** [comment]
> maybe we need to retink the avatar storage location ....

Makes sense. Esepcially if we want to start serving them via dav. And also included trusted owncloud avatars (since we have them in the vcards anyway).

But I think that is not something we can fix for 9.0

**Comment by DeepDiver1975** [comment]
> But I think that is not something we can fix for 9.0

fixing the avatar storage location we cannot fix for 9.0 - but maybe we can get around the skeleton copy operation

**Comment by PVince81** [comment]
Thing is, if the user has no home yet, and we need the avatar, then don't create the skeleton.
Might be possible for the avatar code to detect whether the user has a home yet... but not sure with which API :frowning:

**Comment by rullzer** [comment]
I think I can find some kind of way around this... I'll look into it...

**Comment by rullzer** [comment]
PR in #22410

**Comment by rullzer** [comment]
Issue to rethink how we handle avatars for a later release... https://github.com/owncloud/core/issues/22414

**Comment by PVince81** [comment]
I also discovered that cron.php for CalDAV also does this:
```
0  OC_Util::copySkeleton() /srv/www/htdocs/owncloud/lib/private/util.php:307
1  OC\Files\Node\Root->getUserFolder() /srv/www/htdocs/owncloud/lib/private/files/node/root.php:347
2  OC\AvatarManager->getAvatar() /srv/www/htdocs/owncloud/lib/private/avatarmanager.php:69
3  OC\User\User->getAvatarImage() /srv/www/htdocs/owncloud/lib/private/user/user.php:380
4  OCA\DAV\CardDAV\Converter->getAvatarImage() /srv/www/htdocs/owncloud/apps/dav/lib/carddav/converter.php:164
5  OCA\DAV\CardDAV\Converter->createCardFromUser() /srv/www/htdocs/owncloud/apps/dav/lib/carddav/converter.php:42
6  OCA\DAV\CardDAV\SyncService->updateUser() /srv/www/htdocs/owncloud/apps/dav/lib/carddav/syncservice.php:205
7  OCA\DAV\CardDAV\SyncService->OCA\DAV\CardDAV\{closure}() /srv/www/htdocs/owncloud/apps/dav/lib/carddav/syncservice.php:246
8  OC\User\Manager->callForAllUsers() /srv/www/htdocs/owncloud/lib/private/user/manager.php:337
9  OCA\DAV\CardDAV\SyncService->syncInstance() /srv/www/htdocs/owncloud/apps/dav/lib/carddav/syncservice.php:250
10 OCA\DAV\CardDAV\SyncJob->run() /srv/www/htdocs/owncloud/apps/dav/lib/carddav/syncjob.php:38
11 OC\BackgroundJob\Job->execute() /srv/www/htdocs/owncloud/lib/private/backgroundjob/job.php:52
12 OC\BackgroundJob\TimedJob->execute() /srv/www/htdocs/owncloud/lib/private/backgroundjob/timedjob.php:53
13 {main}          /srv/www/htdocs/owncloud/cron.php:141
```

**Comment by PVince81** [comment]
and copies the skeleton for all my 200+ LDAP users...

**Comment by rullzer** [comment]
This is because the ldap app want the avatar stuff. So hopefully also fixed with #22410
